From: Mykola Golub <mgolub@mirantis.com>
Date: Wed, 18 May 2016 10:07:55 +0300
Subject: rbd-mirror: calculate behind_master only if mirror tag is not newer
 than master

Fixes: http://tracker.ceph.com/issues/15916
Signed-off-by: Mykola Golub <mgolub@mirantis.com>
(cherry picked from commit cbd8d526f94523c9de3c575d426063e63d7e1802)
(cherry picked from commit c3b1bf1e530a4cf55585886a4b4038b9d2862471)

Resolves: rhbz#1340482
---
 src/tools/rbd_mirror/image_replayer/ReplayStatusFormatter.cc | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/tools/rbd_mirror/image_replayer/ReplayStatusFormatter.cc b/src/tools/rbd_mirror/image_replayer/ReplayStatusFormatter.cc
index 303fb58..286a87b 100644
--- a/src/tools/rbd_mirror/image_replayer/ReplayStatusFormatter.cc
+++ b/src/tools/rbd_mirror/image_replayer/ReplayStatusFormatter.cc
@@ -103,7 +103,8 @@ bool ReplayStatusFormatter<I>::calculate_behind_master_or_send_update() {
 
   m_entries_behind_master = 0;
 
-  if (m_master_position == cls::journal::ObjectPosition()) {
+  if (m_master_position == cls::journal::ObjectPosition() ||
+      m_master_position.tag_tid < m_mirror_position.tag_tid) {
     return true;
   }
 
